app/src/main/java/com/example/hellorecycler/MainActivity.kt

```diff
@@ -16,23 +22,35 @@ class MainActivity : AppCompatActivity() {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
-        val myDataset = mutableListOf("Skittles", "Starburst", "Sour Patch Kids")
+        //val myDataset = mutableListOf("Skittles", "Starburst", "Sour Patch Kids")
+
+        // replace with:
+        var myCandyData = ArrayList<CandyData>()
+        myCandyData.add(CandyData(1, "Skittles"))
+        myCandyData.add(CandyData(2, "Starburst"))
+        myCandyData.add(CandyData(3, "Sour Patch Kids"))
 
         viewManager = LinearLayoutManager(this)
-        viewAdapter = MyAdapter(myDataset)
+        viewAdapter = MyAdapter(myCandyData, { candyItem : CandyData -> candyClicked(candyItem) })
```

Comment:
getting "Type mismatch: inferred type is (CandyData) -> Unit but CandyData was expected" here. The lambda expressions seem powerful, but may not have it right

Reply:
you have it defined correctly here, incorrectly in your adapter constructor. I'll leave a comment on that one. 

I am actually super surprised to see you do it this way! Very cool - another (more verbose option) is to create an interface... which you might do if you have multiple callbacks but I like that you only have a function once here.

Reply:
good surprise or bad surprise? :-)